cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -614,6 +617,65 @@ func (s *Server) handleHTTPRoot(w http.ResponseWriter, r *http.Request) {
 	_ = repoTmpl.Execute(w, data)
 }
 
+func (s *Server) handleDebugList(w http.ResponseWriter, r *http.Request) {
+	withIndexed := false
+	indexedStr := r.URL.Query().Get("indexed")
+	if b, err := strconv.ParseBool(indexedStr); err != nil {
+		withIndexed = b
+	}
+
+	var indexed []uint32
+	if withIndexed {
+		indexed = listIndexed(s.IndexDir)
+	}
+
+	repos, err := s.Sourcegraph.List(context.Background(), indexed)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	s.queue.mu.Lock()
+	defer s.queue.mu.Unlock()
+
+	name := ""
+	for _, id := range repos.IDs {
+		if item := s.queue.get(id); item != nil {
+			name = item.opts.Name
+		} else {
+			name = ""
+		}
+
+		_, err := fmt.Fprintf(w, "%d\t%s\n", id, name)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
if you fail to write the use has hung up, so this line doesn't do anything. Normally one will debug log or something.